packages/core/src/createActor.ts

```diff
@@ -207,7 +211,23 @@ export class Actor<TLogic extends AnyActorLogic>
       this.system._set(systemId, this);
     }
 
-    this._initState(options?.snapshot ?? options?.state);
+    this._synchronizer = options?.sync;
+
+    const initialSnapshot =
+      this._synchronizer?.getSnapshot() ?? options?.snapshot ?? options?.state;
+
+    this._initState(initialSnapshot);
+
+    if (this._synchronizer) {
+      this._synchronizerSubscription = this._synchronizer.subscribe(
+        (rawSnapshot) => {
+          const restoredSnapshot =
+            this.logic.restoreSnapshot?.(rawSnapshot, this._actorScope) ??
+            rawSnapshot;
+          this.update(restoredSnapshot, { type: '@xstate.sync' });
```

Comment:
I think it should get enqueued and not handled immediately because with this model we allow for `this.update` to be potentially reentered (and that's a bad thing™). And if we are in the middle of processing 5 events mailbox and we receive a `rawSnapshot` here then likely we should append to the mailbox instead of processing it right away